src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ConflictResolution.targets

```diff
@@ -89,4 +90,49 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   </Target>
 
+  <!--
+    _FilterDuplicateCopyLocalPublishAssets
+    Remove CopyLocalPublishAssets items that could overwrite resolved files to publish.
```

Comment:
So `ResolvedFileToPublish` has precedence over `_ResolvedCopyLocalPublishAssets`?  Why is that?  What do each of these items mean?

Reply:
That's the main reason this PR is a draft, I wanted to check with you which should have precedence. `_ResolvedCopyLocalPublishAssets` consists of assets from packages that need to be copied for publishing and `ResolvedFileToPublish` is eventually everything else, so at this point in execution it contains everything calculated [here](https://github.com/dotnet/sdk/blob/master/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets#L715) which includes child project items, content, resources, compile items, etc.

For example, in the [second test](https://github.com/dotnet/sdk/pull/12085/files#diff-4167a132d5b170df471fdaa1d7da1040R56) System.Runtime.CompilerServices.Unsafe.dll is coming from both the microsoft.netcore.app.runtime.win-x64 package (which is in the `_ResolvedCopyLocalPublishAssets` itemgroup) and it is copied as content in the microsoft.testplatform.cli package (which is originally in the None itemgroup and added to the `ResolvedFileToPublish` itemgroup in [GetCopyToPublishDirectoryItems](https://github.com/dotnet/sdk/blob/master/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets#L715)). 

Here I'm choosing `ResolvedFileToPublish` to have precedence but I'm not confident that this is the right choice.

Reply:
@dsplaisted circling back around to this, do you have any thoughts as to the right behavior here?

As it's getting late in the release cycle and it's possible that there are other cases in which we have duplicte writes that would now produce an error with this change, I'm wondering if it makes more sense to hold off on this and include it in an early preview of 6 so we can get feedback?